src/Collection/Collection.router.ts

```diff
@@ -241,70 +245,76 @@ export class CollectionRouter extends Router {
 
   publishCollection = async (
     req: AuthRequest
-  ): Promise<{ collection: FullCollection; items: FullItem[] }> => {
+  ): Promise<{
+    collection: FullCollection
+    items: FullItem[]
+  }> => {
     const id = server.extractFromReq(req, 'id')
 
-    // We are using the withCollectionExists middleware so we can safely assert the collection exists
-    const [dbCollection, dbItems] = await Promise.all([
-      Collection.findOne<CollectionAttributes>(id),
-      Item.findOrderedByCollectionId(id),
-    ])
-    const remoteCollection = await collectionAPI.fetchCollection(
-      dbCollection!.contract_address!
-    )
-
-    if (!remoteCollection) {
-      // This might be a problem with the graph lagging but we delegate the retry logic on the client
-      throw new HTTPError(
-        'The collection is not published yet',
-        { id },
-        STATUS_CODES.unauthorized
-      )
-    }
-
-    const items: ItemAttributes[] = [...dbItems]
-    let remoteItems: ItemFragment[] = []
-
-    const isMissingBlockchainItemIds = dbItems.some(
-      (item) => item.blockchain_item_id == null
-    )
-
-    if (isMissingBlockchainItemIds) {
-      remoteItems = await collectionAPI.fetchItemsByContractAddress(
-        dbCollection!.contract_address!
-      )
+    try {
+      const dbCollection = await this.service.getDBCollection(id)
 
-      const updates = []
+      let result: {
+        collection: CollectionAttributes
+        items: FullItem[]
+      }
 
-      for (const [index, item] of items.entries()) {
-        const remoteItem = remoteItems.find(
-          (remoteItem) => Number(remoteItem.blockchainId) === index
+      if (isTPCollection(dbCollection)) {
+        const itemIds = server.extractFromReq<string[]>(req, 'itemIds')
+        const dbItems = await Item.findByIds(itemIds)
+        result = await this.service.publishTPCollection(
+          dbCollection,
+          dbItems,
+          server.extractFromReq(req, 'signedMessage'),
+          server.extractFromReq(req, 'signature'),
+          server.extractFromReq(req, 'qty'),
+          server.extractFromReq(req, 'salt')
```

Comment:
just a question: Maybe this should be a single object? both for the client hitting the API but for this method's signature. I'm not sure about naming it tho. I was thinking of `cheque` but not sure.
Not blocking the merge, could be another PR

Reply:
I like the idea! Updated!
`cheque` added to https://github.com/decentraland/builder/pull/1820 as well